qiskit/converters/circuit_to_instruction.py

```diff
@@ -83,10 +83,14 @@ def circuit_to_instruction(circuit, parameter_map=None, equivalence_library=None
     )
     out_instruction.condition = None
 
-    target = circuit.assign_parameters(parameter_dict, inplace=False)
-
-    if equivalence_library is not None:
-        equivalence_library.add_equivalence(out_instruction, target)
+    if equivalence_library is None and not parameter_dict:
+        # avoid making a copy of the circuit
+        # parameter_dict is empty and we do not have to pass a copy to the equivalence library
+        target = circuit
+    else:
+        target = circuit.assign_parameters(parameter_dict, inplace=False)
+        if equivalence_library is not None:
+            equivalence_library.add_equivalence(out_instruction, target)
```

Comment:
This line is separate to whether there are parameters or not, really.  I know adding an equivalence causes a copy, but there's no places in Qiskit that actually use the `equivalence_library` additions (tbh, I'm not 100% sure why we add to the equivalence library at all with the current state of Qiskit).  Minor, but it'd probably be better to move it out of the `else` and have it be its own separate thing; the copy will still be opt-in.